core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -405,6 +415,13 @@ private[spark] class Executor(
           setTaskFinishedAndClearInterruptStatus()
           execBackend.statusUpdate(taskId, TaskState.FAILED, ser.serialize(reason))
 
+        case t: Throwable if task.context.fetchFailed.isDefined =>
```

Comment:
oh, great point!  sorry I missed that.  I've also added a test case for this as well.